src/main/java/com/amplitude/HttpCall.java

```diff
@@ -14,21 +14,30 @@
 
 import java.util.List;
 
-public class HttpCall {
-
+public abstract class HttpCall {
   /*
    * Use HTTPUrlConnection object to make async HTTP request,
    * using data from event like device, class name, event props, etc.
    *
    * @return The response object which contains a code and other information
    */
-  protected static Response syncHttpCallWithEventsBuffer(List<Event> events, String apiKey) {
+  public String apiKey;
+  public List<Event> events;
+  public String apiUrl;
+
+  protected HttpCall(List<Event> events, String apiKey, String apiUrl) {
+    this.apiKey = apiKey;
+    this.events = events;
```

Comment:
Do we need to have the events in constructor? I don't feel we do in this case..
Can we just have apiKey, apiUrl? apiKey is better to be the enum type.
Then the method syncHttpCallWithEventsBuffer will take a list of event as argument.